src/cupbearer/scripts/conf/train_classifier_conf.py

```diff
@@ -64,6 +35,22 @@ def __post_init__(self):
         if isinstance(self.model, (MLP, CNN)):
             self.model.output_dim = self.num_classes
 
+        # For datasets that are not necessarily deterministic based only on
+        # arguments, this is where validation sets are set to follow train_data
+        if hasattr(self.train_data, "backdoor"):
```

Comment:
```suggestion
        if isinstance(self.train_data, BackdoorData):
```
Could we do this instead? (I don't see any reason not to, but might be missing something). That way, we can let the IDE auto-rename things instead of accidentally breaking this without getting a type error.

Reply:
Sounds good. I'll do that.